Building a brand new project (with "zig init") results in recursive panic in docker containers

### Zig Version

0.14.0-dev.1911+3bf89f55c

### Steps to Reproduce and Observed Behavior

## Updated steps to reproduce, based on https://github.com/ziglang/zig/issues/21815#issuecomment-2439785882

`docker run -it alpine:latest`

From within the docker container, install zig, then:

1. `zig init`
2. `zig build`

I believe this is due to the lack of SYS_PTRACE capability (see additional research in comments below), so it may be sensitive to docker version/configuration and Linux kernel version. Steps were performed with Linux 6.1.23 and docker 20.10.17 in its default configuration.

## Original steps to reproduce based on Gitea Actions

1. Create a new zig project with `zig init`
2. Create a basic github actions workflow in `.github/workflows/build.yaml`:
```yaml
name: Basic test
run-name: ${{ github.actor }} building basic zig
on:
  push:
    branches:
      - '*'
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Check out repository code
        uses: actions/checkout@v4
      - name: Setup Zig
        uses: mlugg/setup-zig@v1.2.1
        with:
          version: mach-latest
      - name: Run tests
        run: zig build
```
3. Install [act](https://github.com/nektos/act).
4. Run `act`

## Observed behavior

The zig compiler will successfully build `build.zig`, then will fail running the resulting build executable with a recursive panic. Example output:

```
panic: reached unreachable code
aborting due to recursive panic
error: the following build command crashed:
/home/runner/.cache/zig/o/be1ebd25a51977045011b344ff6125d1/build /opt/hostedtoolcache/zig/0.14.0-dev.1911/x64/zig /opt/hostedtoolcache/zig/0.14.0-dev.1911/x64/lib /home/ec2-user/zig-init /home/runner/.cache/zig /home/runner/.cache/zig --seed 0x272bc170 -Z89a7131e2a7b6f24
```

Note this does not panic on GitHub Actions. This does effect Gitea Actions CI, as it is based on the same runner. Also likely of interest, builds on the same host work just fine. This likely has something to do with the build container configuration.

This was all on linux, tested on kernel 6.1.27 and 6.1.23 with identical results. Zig 0.13.0 works just fine in this environment

### Expected Behavior

Don't panic

--- a/lib/std/debug/MemoryAccessor.zig
+++ b/lib/std/debug/MemoryAccessor.zig
@@ -48,7 +48,8 @@ fn read(ma: *MemoryAccessor, address: usize, buf: []u8) bool {
                 switch (linux.E.init(bytes_read)) {
                     .SUCCESS => return bytes_read == buf.len,
                     .FAULT => return false,
-                    .INVAL, .PERM, .SRCH => unreachable, // own pid is always valid
+                    .INVAL, .SRCH => unreachable, // own pid is always valid
+                    .PERM => {}, // Known to happen in containers.
                     .NOMEM => {},
                     .NOSYS => {}, // QEMU is known not to implement this syscall.
                     else => unreachable, // unexpected

